tests/disaster_recovery_2.py

```diff
@@ -0,0 +1,138 @@
+#
+# Verify that the LIB on A, B, C, and D has stalled and is less than block N. Then, nodes A, B, C, and D can all be
+# cleanly shut down.
+#
+# Then, reversible blocks from all nodes should be removed. All nodes are restarted from an earlier
+# snapshot (prior to block N).
+#
+# P is restarted and replays up to block N after restarting from snapshot. Blocks up to and including block N are sent
+# to the other nodes A, B, C, and D after they are also started up again.
+#
+# Verify that LIB advances and that A, B, C, and D are eventually voting strong on new blocks.
+###############################################################
+
+Print=Utils.Print
+errorExit=Utils.errorExit
+
+args=TestHelper.parse_args({"-d","--keep-logs","--dump-error-details","-v","--leave-running","--unshared"})
+pnodes=1
+delay=args.d
+debug=args.v
+prod_count = 1 # per node prod count
+total_nodes=pnodes+4
+dumpErrorDetails=args.dump_error_details
+
+Utils.Debug=debug
+testSuccessful=False
+
+cluster=Cluster(unshared=args.unshared, keepRunning=args.leave_running, keepLogs=args.keep_logs)
+walletMgr=WalletMgr(True, keepRunning=args.leave_running, keepLogs=args.keep_logs)
+
+try:
+    TestHelper.printSystemInfo("BEGIN")
+
+    cluster.setWalletMgr(walletMgr)
+
+    Print(f'producing nodes: {pnodes}, delay between nodes launch: {delay} second{"s" if delay != 1 else ""}')
+
+    Print("Stand up cluster")
+    specificExtraNodeosArgs={}
+    specificExtraNodeosArgs[0]="--plugin eosio::net_api_plugin --plugin eosio::producer_api_plugin "
+
+    if cluster.launch(pnodes=pnodes, totalNodes=total_nodes, totalProducers=pnodes, specificExtraNodeosArgs=specificExtraNodeosArgs,
+                      topo="./tests/disaster_recovery_2_test_shape.json", delay=delay, loadSystemContract=False,
+                      activateIF=True, signatureProviderForNonProducer=True) is False:
+        errorExit("Failed to stand up eos cluster.")
+
+    assert cluster.biosNode.getInfo(exitOnError=True)["head_block_producer"] != "eosio", "launch should have waited for production to change"
+
+    cluster.biosNode.kill(signal.SIGTERM)
+    cluster.waitOnClusterSync(blockAdvancing=5)
+
+    node0 = cluster.getNode(0) # P
+    node1 = cluster.getNode(1) # A
+    node2 = cluster.getNode(2) # B
+    node3 = cluster.getNode(3) # C
+    node4 = cluster.getNode(4) # D
+
+    Print("Create snapshot (node 0)")
+    ret = node0.createSnapshot()
+    assert ret is not None, "Snapshot creation failed"
+    ret_head_block_num = ret["payload"]["head_block_num"]
+    Print(f"Snapshot head block number {ret_head_block_num}")
+
+    Print("Wait for snapshot node lib to advance")
+    node0.waitForBlock(ret_head_block_num+1, blockType=BlockType.lib)
+    assert node1.waitForLibToAdvance(), "Ndoe1 did not advance LIB after snapshot of Node0"
+
+    assert node0.waitForLibToAdvance(), "Node0 did not advance LIB after snapshot"
```

Comment:
Is this the same as line 87?

Reply:
No, this is just that LIB continues to advance. Probably not really needed, but doesn't hurt to verify.